Source/GameBaseFramework/Public/Interaction/Tasks/GBFAT_WaitForInteractableTargets.h

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "Interaction/GBFInteractionOption.h"
+
+#include <Abilities/Tasks/AbilityTask.h>
+#include <Engine/CollisionProfile.h>
+
+#include "GBFAT_WaitForInteractableTargets.generated.h"
+
+class IGBFInteractableTarget;
+struct FGBFInteractionQuery;
+
+DECLARE_DYNAMIC_MULTICAST_DELEGATE_OneParam( FGBFInteractableObjectsChangedEvent, const TArray< FGBFInteractionOption > &, interactable_options );
+
+UCLASS( Abstract )
+class GAMEBASEFRAMEWORK_API UGBFAT_WaitForInteractableTargets : public UAbilityTask

```

Comment:
could it be possible to inherit from `UGASExtAT_WaitTargetDataHitScan` or another GASExt class?

Reply:
not sure, since those tasks all end after they send the data, and this task keeps running.
We would have to override certain functions and change a bit how this task works, but I don't see much we would gain from doing that.